### PR TITLE
Remove ContainerProxy.

### PR DESCRIPTION
**Pull request by mikeboers**
This was a layer that was once required for garbage collection due to the
cyclic relationships between containers and streams. Newer Cythons seem to deal
with it better.

### CODE REVIEW

**Comment by mikeboers** [comment]
Seems like CI is having a hard time with this. But not because the tests are failing, but not finishing.

**Comment by mikeboers** [comment]
So I think the tests are now passing. Unsure if it is because the various CI was having trouble in general, or if this change causes non-deterministic-ish lockups.

**Comment by jlaine** [comment]
AppVeyor still looks pretty unhappy!

**Comment by jlaine** [comment]
Ah. Looks like this branch needs to be rebased following the flake8 stuff, sorry!

File: av/container/core.pxd [hunk 1]
```diff
@@ -36,27 +30,15 @@ cdef class ContainerProxy(object):
...
-
     cdef readonly ContainerFormat format
 
     cdef readonly dict options
     cdef readonly dict container_options
     cdef readonly list stream_options
 
-    cdef ContainerProxy proxy
-    cdef object __weakref__
+    # cdef object __weakref__
```
Comment (jlaine) [inline av/container/core.pxd]: I think this line can go :)

**Comment by mikeboers** [comment]
Yeah... they all do. That was my biggest holdup. But I figure if I flake8 them first, it won't be too bad.

File: av/stream.pyx [hunk 2]
```diff
@@ -87,10 +86,6 @@ cdef class Stream(object):
         self.codec_context = wrap_codec_context(self._codec_context, self._codec, False)
         self.codec_context.stream_index = stream.index
 
-    def __dealloc__(self):
-        if self._codec_options:
```
Comment (jlaine) [inline av/stream.pyx]: Nice catch, it looks like this was not used at all!

File: av/stream.pxd [hunk 3]
```diff
@@ -3,24 +3,22 @@ from libc.stdint cimport int64_t
...
 from av.frame cimport Frame
 from av.packet cimport Packet
 
 
 cdef class Stream(object):
 
     # Stream attributes.
-    cdef ContainerProxy _container
-    cdef _weak_container
+    cdef readonly Container container
```
Comment (jlaine) [inline av/stream.pxd]: I'm thinking we need some kind of mechanism to break the Container <-> Stream reference loop when the container gets closed, but this can probably be addressed in a separate PR

Comment (mikeboers) [reply av/stream.pxd]: I've been looking into overriding the `tp_traverse` and `tp_clear` so we can have cycles with `__del__` and still deal with it. Not sure if that is something we can actually do, but...

Comment (mikeboers) [reply av/stream.pxd]: I think we could leave the `ContainerProxy` as a sort of `ContainerPointer` that only deals with the de-alloc, which would make the code a LOT simpler... if we removed `Stream.seek`.

(Of the top of my head at least).

**Comment by jlaine** [comment]
> Yeah... they all do. That was my biggest holdup. But I figure if I flake8 them first, it won't be too bad.

Agreed. Let's get this PR merged next as some others will need refactoring afterwards.

**Comment by mikeboers** [comment]
Just pushed the rebase. That was a LOT easier than I thought it would be. I'm now quite happy we pulled the trigger on that flake8 stuff.
